implementations/rust/ockam/ockam_entity/src/lease/json_proto.rs

```diff
@@ -17,18 +22,20 @@ pub enum LeaseProtocolOption {
     Permissions(Vec<LeaseProtocolPermissions>),
 }
 
+//! Top level request structure to initiate creation of a lease.
 #[derive(Serialize, Deserialize, PartialEq, Debug)]
 pub struct LeaseProtocolRequest {
-    action: String,
+    action: String, // create, get, revoke
 
     #[serde(default, skip_serializing_if = "HashMap::is_empty")]
     options: HashMap<String, LeaseProtocolOption>,
 
     #[serde(default, skip_serializing_if = "String::is_empty")]
-    token_id: String,
+    token_id: String, // Lease manager token id for referencing a token. Not the token itself.
 }
 
 impl LeaseProtocolRequest {
+    //! Builds a new lease request of the given action, options and token id. Token is null for create.
```

Comment:
These should be `///` doc comments — `//!` apply to the surrounding declaration (e.g. to the `impl LeaseProtocolRequest` block in this case).